### PR TITLE
MMB-306: Restrict Validity Dates And Tokens To Just Membership Type Certificates

### PR DESCRIPTION
**Pull request by shahrukh-compuco**
## Overview
This pr adjusts validity dates and tokens to be available just for membership type certificates.

## Before
The work was done for all types of certificates.

## After
Its only applicable to membership type certificates.

### CODE REVIEW

File: CRM/Certificate/Service/Certificate.php [hunk 1]
```diff
@@ -25,12 +25,13 @@ public function store($values) {
...
         $params['id'] = $values['id'];
       }
+      $isMembershipCertificate = (int) $values['type'] === CRM_Certificate_Enum_CertificateType::MEMBERSHIPS;
       $params['name'] = $values['name'];
       $params['entity'] = $values['type'];
       $params['end_date'] = $values['end_date'];
       $params['start_date'] = $values['start_date'];
-      $params['min_valid_from_date'] = $values['min_valid_from_date'] ?? NULL;
-      $params['max_valid_through_date'] = $values['max_valid_through_date'] ?? NULL;
+      $params['min_valid_from_date'] = isset($values['min_valid_from_date']) && $isMembershipCertificate ? $values['min_valid_from_date'] : NULL;
```
Comment (olayiwola-compucorp) [inline CRM/Certificate/Service/Certificate.php:33]: @shahrukh-compuco look at how we store extra values for entities here

Example of the event certificate here
https://github.com/compucorp/uk.co.compucorp.certificate/blob/7d66548ebf5e19b6c5f0c477eac90de569063f8f/CRM/Certificate/Service/CertificateEvent.php#L37

For membership certificate, the extra values method will be implemented here

https://github.com/compucorp/uk.co.compucorp.certificate/blob/7d66548ebf5e19b6c5f0c477eac90de569063f8f/CRM/Certificate/Service/CertificateMembership.php

File: js/showHideDateFields.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+/**
```
Comment (olayiwola-compucorp) [inline js/showHideDateFields.js:1]: @shahrukh-compuco see how we update fields based on the selected certificate type here

https://github.com/compucorp/uk.co.compucorp.certificate/blob/7d66548ebf5e19b6c5f0c477eac90de569063f8f/templates/CRM/Certificate/Form/CertificateConfigure.tpl#L64C5-L66C1

I would suggest to keep the code together.

File: CRM/Certificate/Token/Certificate.php [hunk 3]
```diff
@@ -72,63 +71,57 @@ public function prefetch(TokenValueEvent $e) {
...
-
-    if ($membershipStartTimestamp && $membershipEndTimestamp) {
-      $renewalTimestamp = strtotime($membershipDates['end_date'] . " -1 year 1 day");
-      $resolvedTokens['rolling_start_or_renewal_date'] = $renewalTimestamp > $membershipStartTimestamp
-        ? CRM_Utils_Date::customFormat(date('Y-m-d', $renewalTimestamp), '%e/%b/%Y')
-        : CRM_Utils_Date::customFormat($membershipDates['start_date'], '%e/%b/%Y');
+    $resolvedTokens['valid_from'] = '';
+    $resolvedTokens['valid_to'] = '';
+
+    if ((int) $certificate->entity === CRM_Certificate_Enum_CertificateType::MEMBERSHIPS) {
```
Comment (olayiwola-compucorp) [inline CRM/Certificate/Token/Certificate.php:99]: I would suggest logic for Membership based tokens should here:
https://github.com/compucorp/uk.co.compucorp.certificate/blob/7d66548ebf5e19b6c5f0c477eac90de569063f8f/CRM/Certificate/Token/Membership.php

Comment (olayiwola-compucorp) [reply CRM/Certificate/Token/Certificate.php:99]: Also, see how entity tokens are namespaced here

https://github.com/compucorp/uk.co.compucorp.certificate/blob/7d66548ebf5e19b6c5f0c477eac90de569063f8f/certificate.php#L143-L150

**Review by olayiwola-compucorp — CHANGES_REQUESTED** [review]

File: CRM/Certificate/Service/Certificate.php (hunk 1 above)
Comment (shahrukh-compuco) [reply CRM/Certificate/Service/Certificate.php:33]: These are not extra values instead these are certificate properties but its just that they depend on the certificate type to be membership and in storeExtraValues we dont save certificate properties as its already saved before calling the storeExtraValues

File: CRM/Certificate/Token/Certificate.php (hunk 3 above)
Comment (shahrukh-compuco) [reply CRM/Certificate/Token/Certificate.php:99]: as agreed on slack these are actually certificate properties so they make more sense to be in certificate class

File: js/showHideDateFields.js (hunk 2 above)
Comment (shahrukh-compuco) [reply js/showHideDateFields.js:1]: done

**Review by olayiwola-compucorp — APPROVED** [review]

**Pull request merged**